### PR TITLE
Upgrade prettier

### PR DESCRIPTION
**Pull request by kassens**
The old version of prettier we were using didn't support the Flow syntax to access properties in a type using `SomeType['prop']`. This updates `prettier` and `rollup-plugin-prettier` to the latest versions.

I added the prettier config `arrowParens: "avoid"` to reduce the diff size as the default has changed in Prettier 2.0. The largest amount of changes comes from function expressions now having a space. This doesn't have an option to preserve the old behavior, so we have to update this.

### CODE REVIEW

**Comment by kassens** [comment]
Note, I kept the formatting changes in a different commit from manual changes to help review.

File: scripts/prettier/index.js [hunk 1]
```diff
@@ -37,7 +37,7 @@ const files = glob
   .filter(f => !onlyChanged || changedFiles.has(f));
 
 if (!files.length) {
-  return;
+  process.exit(0);
```
Comment (kassens) [inline scripts/prettier/index.js:40]: prettier didn't like the top level return. Used `process.exit(0)` instead to replicate the behavior (see line 82 is also using `process.exit` already).

**Review by gnoff — COMMENTED** [review]
The config changes make sense. the code changes that I checked seem to make sense. What is the expected code review treatment for tool-updated code? I can go line by line but don't want to if not necessary cc @sebmarkbage

**Comment by kassens** [comment]
@gnoff I do a combination of:
- review manual changes (that's why I separated them into separate git commits)
- spot check automated changes that there's no unintended side effects

File: scripts/error-codes/invertObject.js [hunk 2]
```diff
@@ -3,19 +3,16 @@
  *
  * This source code is licensed under the MIT license found in the
  * LICENSE file in the root directory of this source tree.
- *
- * @flow
  */
 'use strict';
 
-import type {ErrorMap} from './Types';
```
Comment (kassens) [inline scripts/error-codes/invertObject.js]: This used a "type comment" before of the form:

```
/*::
[flow only code here]
*/
```
The `flow` prettier parser parses this as a pure AST w/o the comment and then when it's printed the comment is removed. Most of the scripts don't use types, I think we don't loose much by removing this here.

File: .prettierrc.js [hunk 3]
```diff
@@ -8,8 +8,8 @@ module.exports = {
   jsxBracketSameLine: true,
   trailingComma: 'es5',
   printWidth: 80,
-  parser: 'babel',
-
+  parser: 'flow',
```
Comment (gnoff) [inline .prettierrc.js:11]: should this be 'hermes' or 'flow'?

Comment (kassens) [reply .prettierrc.js:11]: The hermes PR wasn't merged yet due to concerns of size IIRC. Looks like flow is current enough for the moment to parser everything. (not actually sure what the plan is going forward with hermes vs. flow parser).

Comment (gnoff) [reply .prettierrc.js:11]: Ah I see

**Comment by gnoff** [comment]
@kassens When I try to annotate the type of `this` for functions the `flow` parser chokes but the `babel` parser still works

```
function myFunc (this: Thing) {}

let thing: Thing = {};
myFunc.call(thing);
```

This flow syntax is a little new to me, not sure when they introduced it but seems necessary without suppressing warnings and it's nice to have some type safety for the this argument. Is there a reason `flow` is better than `babel` for the prettier parser?

Also, I tried something like this with both `flow` and `babel` parsers and neither supported it

```
type X<T: ['one', 'two']> = {
  one: T[0],
  two: T[1]
}
```

Maybe this is still blocked on the hermes-flow parser patch?

**Pull request merged**